src/libcore/str/lossy.rs

```diff
@@ -17,6 +15,7 @@ impl Utf8Lossy {
     }
 
     pub fn from_bytes(bytes: &[u8]) -> &Utf8Lossy {
+        // SAFETY: Both use the same memory layout, and UTF-8 correctness isn't required.
```

Comment:
FWIW, they actually are not guaranteed to use the same memory layout -- `Utf8Lossy` has no `repr` annotations. In this case I think adding `repr(transparent)` to `Utf8Lossy` is enough.

If you know of more such cases, it would be great if you could add them in https://github.com/rust-lang/unsafe-code-guidelines/issues/90. :)